### ISSUE TITLE
google-bigquery and pubsub tests fail in native mode on Camel 4

### ISSUE THREAD

**Issue by ppalaga** [apache/camel-quarkus#4502]
Both fail due to 
```
io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory the class was requested to be initialized at run time
```

The whole message:

```
[INFO] [stdout] GraalVM Native Image: Generating 'camel-quarkus-integration-test-google-pubsub-3.0.0-SNAPSHOT-runner' (executable)...
[INFO] [stdout] ========================================================================================================================
[INFO] [stdout] [1/7] Initializing...                                                                                   (18.7s @ 0.61GB)
[INFO] [stdout]  Version info: 'GraalVM 22.3.0 Java 17 CE'
[INFO] [stdout]  Java version info: '17.0.5+8-jvmci-22.3-b08'
[INFO] [stdout]  C compiler: gcc (redhat, x86_64, 11.3.1)
[INFO] [stdout]  Garbage collector: Serial GC
[INFO] [stdout]  8 user-specific feature(s)
[INFO] [stdout]  - com.google.api.gax.grpc.nativeimage.GrpcNettyFeature
[INFO] [stdout]  - com.google.api.gax.grpc.nativeimage.ProtobufMessageFeature
[INFO] [stdout]  - com.google.api.gax.nativeimage.GoogleJsonClientFeature
[INFO] [stdout]  - com.google.api.gax.nativeimage.OpenCensusFeature
[INFO] [stdout]  - com.oracle.svm.thirdparty.gson.GsonFeature
[INFO] [stdout]  - io.quarkus.runner.Feature: Auto-generated class by Quarkus from the existing extensions
[INFO] [stdout]  - io.quarkus.runtime.graal.DisableLoggingFeature: Disables INFO logging during the analysis phase for the [org.jboss.threads] categories
[INFO] [stdout]  - io.quarkus.runtime.graal.ResourcesFeature: Register each line in META-INF/quarkus-native-resources.txt as a resource on Substrate VM
[INFO] [stdout] Warning: RecomputeFieldValue.FieldOffset automatic substitution failed. The automatic substitution registration was attempted because a call to sun.misc.Unsafe.objectFieldOffset(Field) was detected in the static initializer of com.google.protobuf.UnsafeUtil. Detailed failure reason(s): The argument of Unsafe.objectFieldOffset(Field) is not a constant field., Could not determine the field where the value produced by the call to sun.misc.Unsafe.objectFieldOffset(Field) for the field offset computation is stored. The call is not directly followed by a field store or by a sign extend node followed directly by a field store. 
[INFO] [stdout] [2/7] Performing analysis...  []                                                                        (27.2s @ 2.19GB)
[INFO] [stdout]   16,773 (85.10%) of 19,710 classes reachable
[INFO] [stdout]   22,364 (62.28%) of 35,909 fields reachable
[INFO] [stdout]   71,885 (47.66%) of 150,826 methods reachable
[INFO] [stdout]      827 classes, 1,070 fields, and 11,793 methods registered for reflection
[INFO] [stdout] 
[WARN] [stderr] com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
[WARN] [stderr]  io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory the class was requested to be initialized at run time (from 'META-INF/native-image/com.google.api/gax-grpc/native-image.properties' in 'file:///home/ppalaga/orgs/cq/camel-quarkus/integration-tests/google-pubsub/target/camel-quarkus-integration-test-google-pubsub-3.0.0-SNAPSHOT-native-image-source-jar/lib/com.google.api.gax-grpc-2.18.1.jar' with 'io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory'). To see why io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory got initialized use --trace-class-initialization=io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory
[WARN] [stderr] To see how the classes got initialized, use --trace-class-initialization=io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory
[WARN] [stderr] Error: Classes that should be initialized at run time got initialized during image building:
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:73)
[WARN] [stderr]  io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory the class was requested to be initialized at run time (from 'META-INF/native-image/com.google.api/gax-grpc/native-image.properties' in 'file:///home/ppalaga/orgs/cq/camel-quarkus/integration-tests/google-pubsub/target/camel-quarkus-integration-test-google-pubsub-3.0.0-SNAPSHOT-native-image-source-jar/lib/com.google.api.gax-grpc-2.18.1.jar' with 'io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory'). To see why io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory got initialized use --trace-class-initialization=io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ProvenSafeClassInitializationSupport.checkDelayedInitialization(ProvenSafeClassInitializationSupport.java:273)
[WARN] [stderr] To see how the classes got initialized, use --trace-class-initialization=io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory
[INFO] [stdout] ------------------------------------------------------------------------------------------------------------------------
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.duringAnalysis(ClassInitializationFeature.java:164)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$10(NativeImageGenerator.java:748)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:85)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:535)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$11(NativeImageGenerator.java:748)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
[WARN] [stderr]         at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:162)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:580)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:745)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
[WARN] [stderr]         at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:578)
[INFO] [stdout]                         2.6s (5.6% of total time) in 28 GCs | Peak RSS: 6.51GB | CPU load: 7.24
```

**Comment by JiriOndrusek** [comment]
Fix is prepared in https://github.com/apache/camel-quarkus/pull/4546 and requires camel change - https://github.com/apache/camel/pull/9326

**Comment by ppalaga** [comment]
Fixed via d28d5a06f17be1648bcb2bd019dd41891f85f92c
